torch/distributed/fsdp/_fully_shard/_fully_shard.py

```diff
@@ -91,6 +91,7 @@ def fully_shard(
     mp_policy: MixedPrecisionPolicy = MixedPrecisionPolicy(),
     offload_policy: OffloadPolicy = OffloadPolicy(),
     ignored_params: Optional[set[nn.Parameter]] = None,
+    allocate_memory_from_process_group: bool = False,
```

Comment:
So if I understand correctly this is just for collective buffers right? Does it make sense to move this call from `fully_shard` to `FSDPModule` like

```
m: FSDPModule = fully_shard(...)
m.set_allocate_memory_from_process_group_for_comm(True)
```

?

This is to differentiate from the regular tensor alloc during sharding in constructor.

@lw @weifengpy @kwen2501 lmk if that makes sense?